.github/workflows/release-v2.yml

```diff
@@ -128,45 +149,190 @@ jobs:
+                <password>${ARTIFACTORY_PASSWORD}</password>
+              </server>
+              <server>
+                <id>artifactory-releases</id>
+                <username>${ARTIFACTORY_USERNAME}</username>
+                <password>${ARTIFACTORY_PASSWORD}</password>
+             </server>
+           </servers>
+          </settings>
+          EOF
+
+      - name: Install coordinator
+        run: |
+          JAVA_HOME=$JAVA_8 ./mvnw versions:set -DremoveSnapshot versions:commit
+          JAVA_HOME=$JAVA_8 ./mvnw -B clean install -P dse -DskipTests 
+
+      # only set version here
+      - name: Install APIs
         run: |
-          mvn versions:set -DremoveSnapshot versions:commit && \
-          mvn -B -Dgpg.passphrase=${{ secrets.OSSRH_GPG_SECRET_KEY_PASSWORD }} clean deploy -DskipTests -P deploy,dse
+          cd apis/
+          JAVA_HOME=$JAVA_17 ./mvnw versions:set -DremoveSnapshot versions:commit
+          cd ../
+
       - name: Set up Docker QEMU
-        uses: docker/setup-qemu-action@v1
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      # build and push images to Docker buh
       - name: Login to DockerHub
-        uses: docker/login-action@v1
+        uses: docker/login-action@v2
         with:
           username: ${{ secrets.DOCKER_USERNAME }}
           password: ${{ secrets.DOCKER_TOKEN }}
-      - name: Set up Docker Buildx
-        id: buildx
-        uses: docker/setup-buildx-action@v1
-        with:
-          install: true
-      - name: Build and push
+
+      - name: Build and push (coordinator, DockerHub)
+        if: ${{ inputs.publish != false }}
         run: |
           ./build_docker_images.sh -p -t ${{ steps.vars.outputs.tag }}
+
+      - name: Build and push (apis, DockerHub)
+        if: ${{ inputs.publish != false }}
+        run: |
+          cd apis/
+          JAVA_HOME=$JAVA_17 ./mvnw clean package -DskipTests -Dquarkus.container-image.build=true -Dquarkus.container-image.push=true -Dquarkus.container-image.tag=${{ steps.vars.outputs.tag }}
+          cd ../
+
+      # repeat the same for the AWS ECR
       - name: Configure AWS credentials
         uses: aws-actions/configure-aws-credentials@v1
         with:
           aws-access-key-id: ${{ secrets.ECR_ACCESS_KEY }}
           aws-secret-access-key: ${{ secrets.ECR_SECRET_KEY }}
           aws-region: us-east-1
+
       - name: Login to Amazon ECR
         id: login-ecr
         uses: aws-actions/amazon-ecr-login@v1
-      - name: Build and push (ECR)
+
+      - name: Build and push (coordinator, Amazon ECR)
+        if: ${{ inputs.publish != false }}
         run: |
           ./build_docker_images.sh -p -t ${{ steps.vars.outputs.tag }} -r ${{ secrets.ECR_REPOSITORY }} -a
-      - name: Update version numbers for next release
+
+      - name: Build and push (apis, Amazon ECR)
+        if: ${{ inputs.publish != false }}
```

Comment:
`${{ inputs.publish == true }}` seems easier to understand compared to `${{ inputs.publish != false }}`. ` `${{ inputs.publish }}` not work, or does it need `!=` (or `==`) for coercion?

Reply:
ah, nm, I understand, this work well for testing. :) Because absence is true.

Reply:
But how would this work when it's not a manual trigger and there are no inputs?

Reply:
I didn't see your response. I understand, the absence is true to allow for testing.